downstairs/src/lib.rs

```diff
@@ -1275,7 +1313,7 @@ impl Downstairs {
     }
 
     /*
-     * Only grab the lock if the Upstairs UUID matches.
+     * Only grab the lock if the Upstairs connection matches.
```

Comment:
I think "connection" here might be confusing, as it's a generic enough name and a reader might get confused.  What if instead this was `UpstairsConnection`.  This might signal to the reader that there are specific things that are checked and gives a little more detail.

Reply:
Tried to clarify and re-word in f37ab1a, let me know what you think.